addons/project/project_view.xml

```diff
@@ -111,12 +111,12 @@
                     <notebook>
                         <page string="Settings">
                             <group>
-                                <field name="partner_id" on_change="onchange_partner_id(partner_id)" string="Customer"/>
+                                <field name="privacy_visibility" widget="radio"/>
+                                <field name="partner_id" on_change="onchange_partner_id(partner_id)" string="Customer" attrs="{'invisible': [('privacy_visibility', '!=', 'portal')]}"/>
```

Comment:
Whu is the customer field hidden when not in portal ? You could have a private (followers only) with a customer.